src/autocomplete/NotifProvider.js

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import AutocompleteProvider from './AutocompleteProvider';
+import { _t } from '../languageHandler';
+import MatrixClientPeg from '../MatrixClientPeg';
+import {PillCompletion} from './Components';
+import sdk from '../index';
+
+const AT_ROOM_REGEX = /@\S*/g;
```

Comment:
I guess this means that users starting with the first part (of any length) of `@room` will be slightly more difficult to mention?

Reply:
Yeah, but not sure what we can do about this short of changing it to *room or something

Reply:
Well we could make it `/@room/`, mandating an entire match, so you'd have to type the whole shebang and then press tab

Reply:
That would still cause problems for people like late-noughties hitmakers [@room5](https://www.youtube.com/watch?v=UcHs2f2TSxI).

Reply:
Fair enough!